src/inverted-index.js

```diff
@@ -146,33 +139,42 @@ class InvertedIndex {
    * @returns{Object} - The results of the search
    */
   searchIndex(...args) {
-    let searchResults = {};
+    const allArgs = args.toString().toLowerCase().split(',');
+    const searchResults = {};
+    let filesToSearch;
 
     /**
-     * This neat trick allows us to 'flatten' an array
-     * ...args is an array that contains multiple arguments
-     * Convert allArgs to lower case /
+     * Query is an array that contains the terms we want to search for
+     * It is simply a copy of the allArgs array
      */
-    let allArgs = args.toString().toLowerCase();
-
-    allArgs = allArgs.split(',');
-
-    // Get all the created indexes
-    const allIndexes = this.getIndex();
+    const query = allArgs.slice();
 
-    // Check if the first element of array is not a .json file
-    if (args[0].slice(-5) !== '.json') {
-      searchResults = InvertedIndex.iterCollection(allArgs, allIndexes);
-    } else if (Object.keys(allIndexes).indexOf(allArgs[0]) === -1) {
-      return ('Index for file not found!');
+    if (allArgs[0].slice(-5) === '.json') {
+      /**
+       * Assign filesToSearch to the first element of allArgs if it is a
```

Comment:
### Critical

This is to notify you, so you understand that you actually meant the polar opp of what this comment says.